src/pages/ProfilesExplorerView/domain/variables/FiltersVariable/FiltersVariable.tsx

```diff
@@ -4,83 +4,77 @@ import { useStyles2 } from '@grafana/ui';
 import { CompleteFilters } from '@shared/components/QueryBuilder/domain/types';
 import { QueryBuilder } from '@shared/components/QueryBuilder/QueryBuilder';
 import { useQueryFromUrl } from '@shared/domain/url-params/useQueryFromUrl';
-import React, { useEffect, useMemo } from 'react';
+import React, { useEffect } from 'react';
 
-import { ProfileMetricVariable } from '../ProfileMetricVariable';
+import { useBuildPyroscopeQuery } from '../../useBuildPyroscopeQuery';
 import { ProfilesDataSourceVariable } from '../ProfilesDataSourceVariable';
-import { ServiceNameVariable } from '../ServiceNameVariable';
-import { convertPyroscopeToVariableFilter, expressionBuilder } from './filters-ops';
+import { convertPyroscopeToVariableFilter } from './filters-ops';
 
 export class FiltersVariable extends AdHocFiltersVariable {
   static DEFAULT_VALUE = [];
 
-  constructor() {
+  constructor({ key }: { key: string }) {
     super({
-      key: 'filters',
-      name: 'filters',
+      key,
+      name: key,
       label: 'Filters',
       filters: FiltersVariable.DEFAULT_VALUE,
+      expressionBuilder: (filters) =>
```

Comment:
Native is always better :) This will help us for https://github.com/grafana/explore-profiles/pull/122